common/src/test/java/io/netty/util/concurrent/DefaultThreadFactoryTest.java

```diff
@@ -0,0 +1,243 @@
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.netty.util.concurrent;
+
+import org.junit.Test;
+
+import java.security.Permission;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.concurrent.atomic.AtomicReference;
+
+import static org.hamcrest.CoreMatchers.equalTo;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.junit.Assert.assertNull;
+
+public class DefaultThreadFactoryTest {
+
+    // test that ensures DefaultThreadGroup is not sticky about thread groups by default
+    @Test(timeout = 2000)
+    public void testStickyThreadGroups() throws InterruptedException {
+        final AtomicReference<DefaultThreadFactory> factory = new AtomicReference<DefaultThreadFactory>();
+        final AtomicReference<ThreadGroup> capturedGroup = new AtomicReference<ThreadGroup>();
+        final AtomicReference<Thread.UncaughtExceptionHandler> capturedUh =
+                new AtomicReference<Thread.UncaughtExceptionHandler>();
+
+        final ThreadGroup brother = new ThreadGroup("brother");
+
+        final Thread first = new Thread(brother, new Runnable() {
+            @Override
+            public void run() {
+                factory.set(new DefaultThreadFactory(this.getClass()));
+            }
+        });
+        first.start();
+        first.join();
+
+        final AtomicReference<Throwable> interrupted = new AtomicReference<Throwable>();
+
+        final ThreadGroup sister = new ThreadGroup("sister");
+        final Thread second = new Thread(sister, new Runnable() {
+            @Override
+            public void run() {
+                final Thread t = factory.get().newThread(new Runnable() {
+                    @Override
+                    public void run() {
+                        capturedGroup.set(Thread.currentThread().getThreadGroup());
+                        capturedUh.set(Thread.currentThread().getUncaughtExceptionHandler());
+                    }
+                });
+                t.start();
+                try {
+                    t.join();
+                } catch (InterruptedException e) {
+                    interrupted.set(e);
+                    Thread.currentThread().interrupt();
+                }
+            }
+        });
+
+        assertNull(interrupted.get());
+
+        second.start();
+        second.join();
+
+        assertThat(capturedGroup.get(), equalTo(sister));
+        assertThat(capturedUh.get(), equalTo((Thread.UncaughtExceptionHandler) sister));
+    }
+
+    @Test(timeout = 2000)
+    public void testDescendantThreadGroups() throws InterruptedException {
+        SecurityManager current = null;
+
+        try {
+            current = System.getSecurityManager();
```

Comment:
@Scottmitch Sure, I don't think it's relying on implementation details. The Javadocs do say:

```java
/**
     * Gets the system security interface.
     *
     * @return  if a security manager has already been established for the
     *          current application, then that security manager is returned;
     *          otherwise, <code>null</code> is returned.
     * @see     #setSecurityManager
     */
```

with no `@throws` section. That said, moving the assignment outside of the try block has the advantage of permitting `current` to be modified as `final`. I made the change in db9d3b46149bf659a1cbec99a5977cd06b30cc02.